src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -2130,6 +2131,17 @@ void addSymbol(Symbol sym, LambdaSymbolKind skind) {
                 }
             }
 
+            private boolean outerThisReachable(Type current, Symbol target) {
```

Comment:
The idea here is: if we can reach the target symbol from the current one w/o traversing any type under construction, or any type for which `NOOUTERTHIS` is set, then we can consider the target `T.this` reachable.